network-api/networkapi/wagtailpages/templatetags/mini_site_tags.py

```diff
@@ -23,16 +24,21 @@ def mini_site_horizontal_nav(context, page):
 # Render a page's CTA (petition, signup, etc.)
 @register.inclusion_tag("wagtailpages/tags/cta.html", takes_context=True)
 def cta(context, page):
-    cta = {
-        "page": page,
-        "cta": None,
-        "cta_type": None,
-        "lang": context["request"].LANGUAGE_CODE,
-        "source_url": context["request"].build_absolute_uri(),
-        "thank_you_url": context["request"].build_absolute_uri(context["request"].path + "?thank_you=true"),
-        "show_formassembly_thank_you": context["request"].GET.get("thank_you") == "true",
-        "csp_nonce": context["request"].csp_nonce,
-    }
+
+    def generate_thank_you_url(url):
```

Comment:
Is there a simpler implementation for this helper function `generate_thank_you_url`? I was thinking to just test on the existence of `?` and append `thank_you=true` accordingly lol

Reply:
The `parsed_url` seems fine, I just don't like the `"&".join`. We can get the query params as a dictionary by using:

```python
from urlib.parse import parse_qs
query_params = parse_qs(parsed_url.query)
```

Then we can append `thank_you=True` and rebuild the url. Putting it all together, I would probably re-write it as:

```python
parsed_url = urlparse(url)
query_params = parse_qs(parsed_url.query)
query_params["thank_you"] = True
query_string = urlencode(query_params)
thank_you_url = urlunparse(parsed_url._replace(query=query_string))
```

Reply:
Thanks for the guidance, @jhonatan-lopes ! I didn't really like using `.join` either but wasn't able to look up for a smarter implementation lol.

Note that I ended up using `parse_qsl` as the format it returns for the key-value pairs works better here 😁 .